code/modules/antagonists/traitor/objectives/sabotage_machinery.dm

```diff
@@ -0,0 +1,235 @@
+		possible_jobs -= objective.chosen_job
+	for(var/available_job in possible_jobs)
+		var/job_machine_path = possible_jobs[available_job]
+		if (!length(GLOB.objective_machine_handler.machine_instances_by_path[job_machine_path]))
+			possible_jobs -= available_job
+	if(!length(possible_jobs))
+		return FALSE
+
+	chosen_job = pick(possible_jobs)
+	var/list/obj/machinery/possible_machines = GLOB.objective_machine_handler.machine_instances_by_path[possible_jobs[chosen_job]]
+	for(var/obj/machinery/machine as anything in possible_machines)
+		prepare_machine(machine)
+
+	replace_in_name("%JOB%", lowertext(chosen_job))
+	replace_in_name("%MACHINE%", possible_machines[1].name)
+	return TRUE
+
+/// Marks a given machine as our target
+/datum/traitor_objective/sabotage_machinery/proc/prepare_machine(obj/machinery/machine)
+	AddComponent(/datum/component/traitor_objective_register, machine, succeed_signals = list(COMSIG_PARENT_QDELETING))
+
+// Destroy machines which are in annoying locations, are annoying when destroyed, and aren't directly interacted with
+/datum/traitor_objective/sabotage_machinery/destroy
+	name = "Destroy the %MACHINE%"
+	description = "Destroy the %MACHINE% to cause disarray and disrupt the operations of the %JOB%'s department."
+
+	progression_reward = list(5 MINUTES, 10 MINUTES)
+	telecrystal_reward = list(3, 4)
+
+	progression_minimum = 15 MINUTES
+	progression_maximum = 30 MINUTES
+
+	applicable_jobs = list(
+		JOB_STATION_ENGINEER = /obj/machinery/telecomms/hub,
+		JOB_SCIENTIST = /obj/machinery/rnd/server,
+	)
+
+// Rig machines which are in public locations to explode when interacted with
+/datum/traitor_objective/sabotage_machinery/trap
+	name = "Sabotage the %MACHINE%"
+	description = "Destroy the %MACHINE% to cause disarray and disrupt the operations of the %JOB%'s department. If you can get another crew member to destroy the machine using the provided booby trap, you will be rewarded with an additional %PROGRESSION% reputation and %TC% telecrystals."
+
+	progression_reward = list(2 MINUTES, 8 MINUTES)
+	telecrystal_reward = 0 // Only from completing the bonus objective
+
+	progression_minimum = 0 MINUTES
+	progression_maximum = 10 MINUTES
+
+	maximum_allowed = 2
+	applicable_jobs = list(
+		JOB_CHIEF_ENGINEER = /obj/machinery/rnd/production/protolathe/department/engineering,
+		JOB_CHIEF_MEDICAL_OFFICER = /obj/machinery/rnd/production/techfab/department/medical,
+		JOB_HEAD_OF_PERSONNEL = /obj/machinery/rnd/production/techfab/department/service,
+		JOB_QUARTERMASTER = /obj/machinery/rnd/production/techfab/department/cargo,
+		JOB_RESEARCH_DIRECTOR = /obj/machinery/rnd/production/protolathe/department/science,
+		JOB_SHAFT_MINER = /obj/machinery/mineral/ore_redemption,
+	)
+
+	/// Bonus reward to grant if you booby trap successfully
+	var/bonus_tc = 2
+	/// Bonus reputation to grant if you booby trap successfully
+	var/bonus_rep = 2 MINUTES
```

Comment:
I think these should be variable. Honestly, the bonus rep is kinda disappointing.
Can you instead set the progression_reward to something very low and beef up the bonus_rep?
Also don't call it rep, call it progression in code. It's important to remember that reputation is the visual representation of progression. Progression is the time representation of reputation.